Tidy up susecon work

KlapTrap opened the pull request.

nwmac committed:
WIP: Kube dashboard proxy

Commit:
Move tabs to left nav

Commit:
Add page header from side nav

Commit:
Add new page side nav

Commit:
Remove old side nav open close logic

Commit:
Improve sub nav layout

Commit:
Remove unused mobile nav logic

Commit:
Add icons to cf page

Commit:
Update icons for all CF side navs

Commit:
Tidy up page sub nav

Commit:
Fix scss import

Commit:
Merge branch 'v2-master' into tabs-layout-change

* v2-master:
  Fix test failure on faster system
  Remove empty rule
  Address review feedback
  Delete secrets.yaml.sys
  Fix e2e tests
  Tidy up a tad
  WIP
  Resolve conflicts. Fit in with customization
  Hide app vars tab if user is not a space developer
  Lint fix & column rename
  Show app chip list for space routes list bound apps - As per space service instance list - Contributes to #2610
  Update side-nave logo with light version

Commit:
Merge branch 'v2-master' into tabs-layout-change

* v2-master: (27 commits)
  Update status_updates.md
  Unit test fixes
  Added focus to first steps, ensure forms are common width
  Removed second entity.entity.entity check
  Fix weird search/replace issue
  Merge upstream changes properly
  Fix cancel on redeploy, removed some more github specific naming
  Fix & optimise e2e test
  Address PR Feedback
  wip
  General improvements - Fixed appFocus directive - Set focus of confirm dialog to cancel instead of confirm - Add/Edit App name - Add more space between async name taken check spinner and input field - Tidied up imports - Removed unused `AppEventDetailDialogComponentComponent`. Looks like a bad merge remnent - Fixed `newNonnAngularInterval` typo
  Ensure we don't try to get length of undefined endpoint description
  Remove global manage apps link - also removed the now defunct select users first step
  Fix e2e test
  Fix e2e test
  E2E Fix
  Fix e2e tests
  Fix e2e tests for GitLab changes
  Unit test fixes
  Fix unit tests
  ...

Commit:
Remove manage user button

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into kube-dashboard

Commit:
Merge branch 'v2-master' into tabs-layout-change

* v2-master: (415 commits)
  CC Fix
  CC Fixed and copy change
  Update fetch all users max pagination
  Merge fixes
  Minor text change
  Small text change
  Minor change to the invite table header
  Update invite users text
  Update the docs guide
  Update docs
  Fix unit test
  Fix imports
  Publish release pipeline fix (#3404)
  Re-instate setup mode
  Weekly update
  Ensure we don't add invalid state when reseting pagination
  Add create-many-users script
  Automated test script improvements and fixes (#3402)
  Fixed date --> data typo
  Remove TODOs for now
  ...

Commit:
Fix home left tab

Commit:
Add WIP breadcrumbs

nwmac committed:
Merge remote-tracking branch 'origin/monorepo' into v2-master

nwmac committed:
Helm Chart support (Monocular integration)

nwmac committed:
Merge branch 'v2-master' of github.com:SUSE/stratos into v2-master

nwmac committed:
Fix tidy up bug

nwmac committed:
Merge remote-tracking branch 'origin/kube-dashboard' into helm

nwmac committed:
Update icon font

nwmac committed:
Merge remote-tracking branch 'origin/merge-upstream' into v2-master

nwmac committed:
Merge remote-tracking branch 'origin/endpoint-cards' into v2-master

nwmac committed:
All updates

nwmac committed:
Update sub type for GKE

nwmac committed:
Merge all latest PRs

nwmac committed:
Merge remote-tracking branch 'upstream/icons-update' into helm-latest

nwmac committed:
Update icons

nwmac committed:
Merge endpoint connect help

nwmac committed:
Latest and greatest

nwmac committed:
Kube dashboard

Commit:
Merge branch 'v2-master' into tabs-layout-change

* v2-master: (130 commits)
  docs update
  Fix unit test
  Fix issues from review - Ensure extensions/extension-types.ts remains free from other imports - Now check endpoint detail type is correct at runtime - Cf Endpoint Type is now registered in cf package - Ensure cf package tests & lint execute correctly - Ensure endpoint details components are destroyed correctly
  Fix merge issue
  Fix merge issue
  More info
  Fix lint issues
  Add note
  Fix missing .entity
  Priorities & tweaks
  Remove index.html file
  Add support for endpoint sub type (and metadata) (#3423)
  Fix service broker action - Ensure that the guid property is properly set - This means error handling happens correcty
  Update cf-api-v3.md
  Updaes
  Another lint fix
  Fix linting, don't show image if no image
  Disable endpoint card routerLink using new directive
  Missing images
  Fix existing getIconButton bug - There may be other places where this magically worked and now correctly won't
  ...

Commit:
Post merge fixes

nwmac committed:
Fixes

nwmac committed:
Updates

Commit:
Editor config change

Commit:
Applciation sub nav and sub nav changes

nwmac committed:
Fix backend build script

Commit:
Fix helm module

Commit:
Add favorite to kube endpoint

Commit:
Fix endpoint card favorite star

Commit:
Remove unneeded favorite config

nwmac committed:
Add API to determine if kube dashboard is installed

nwmac committed:
Merge branch 'helm-latest' of github.com:SUSE/stratos into helm-latest

nwmac committed:
Fix compilation issues

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into tabs-layout-change

nwmac committed:
Merge branch 'fix-unknown-endpoint-types' into tabs-layout-change

nwmac committed:
Fix for endpoint favorite card when endpoint type is not known

nwmac committed:
Add support for icon mode and a few tidy ups

Commit:
Add base kubernetes summary view

Commit:
WIP Kube summary charts

nwmac committed:
Various fixes and tweaks

Commit:
Added simple usage chart and pod usage

Commit:
Add status graphs

Commit:
Little bit of tidy up on kube summary page

Commit:
Tidy up kube summary

nwmac committed:
Merge remote-tracking branch 'origin/v2-master' into navigation-change

nwmac committed:
Fix code climate issues

nwmac committed:
Fix lint issue

nwmac committed:
Latest updates

nwmac committed:
Merge remote-tracking branch 'upstream/navigation-change' into helm-latest

nwmac committed:
Add tab labels and icons

nwmac committed:
Fix merge conflicts + tweaks

nwmac committed:
Fix delete button missing and missing icon

nwmac committed:
PR feedback fixes

nwmac committed:
Status updates

nwmac committed:
PR Fixes

richard-cox committed:
Remove warning on startup

richard-cox committed:
Changes following review

nwmac committed:
Fixes and updates

nwmac committed:
Fix compilation errors

richard-cox committed:
Temp fix for tables with cells that use `white-space: nowrap;` and have a loooong words

richard-cox committed:
Merge remote-tracking branch 'upstream/navigation-change' into susecon

richard-cox committed:
Merge fixes

richard-cox committed:
chip pills are now not bold

richard-cox committed:
Endpoint status updates

richard-cox committed:
Disable connect of endpoint types that don't support it

richard-cox committed:
Sidenav icon

richard-cox committed:
Add helm status to endpoint lists
- Note - we may want to find another pluggable way of showing custom status text/icons

richard-cox committed:
Move view namespace in dashboard link to own column, only show if we have the dashboard

richard-cox committed:
Add loading indicator to k8s summary page to avoid graph flicker

richard-cox committed:
Split out yaml overrides into second step in install helm chart
- moved step1 into core step as there's not a lot of code
- could also split out name/namespace

richard-cox committed:
Update all metrics icons to `equalizer`

richard-cox committed:
Release status & ensure all GetAllEndpoints actions result in a GetSystemSuccess
- use same endpoint status on helm/releases as endpoint list
- we were using the GetAllEndpoints action when working with lists of endpoints..
however this actually did nothing. Now we fire off a GetSystemSuccess whenever
we dispatch GetAllEndpoints

richard-cox committed:
Remove pod image & tag columns, add `Ready Containers` using `containerStatuses`

nwmac committed:
Various updates

nwmac committed:
Fix endpoint notification. Tidy up logging

richard-cox committed:
Add helm versions list to helm config tab

richard-cox committed:
Also show 'not installed' in helm config versions table

nwmac committed:
Fix release resource parsing

nwmac committed:
Merge branch 'susecon' of github.com:SUSE/stratos into susecon

richard-cox committed:
Add pods page & list to helm release page

richard-cox committed:
Add majority of helm release services page & list
Only thing missing should be the populate of fields in the card, the data is available though just a bit of wiring needed

richard-cox committed:
susecon script

Commit:
Tidy up kube summary page

richard-cox committed:
Tweaks/Fixes to Helm Release Services

richard-cox committed:
Tidy up release loading indicators

Commit:
Ensure all padding is consistent with dashboard component

richard-cox committed:
Add services view to kubernetes namespace page
- to do so made base kube services list config and data source classes
- Tweaked kuberetes 'application' name to be inline with labels on how to get there

Commit:
Add base step to application creation.

Commit:
CC Fixes

Commit:
E2e fixes

Commit:
Unit test fixes

richard-cox committed:
Minor improvements regarding kube/helm row uniqueness. Handle helm errors better

richard-cox committed:
Show helm release service cards quicker

richard-cox committed:
Only show install button if we have connected k8s & only show connected k8s in install stepper

Commit:
Unit test fix

richard-cox committed:
Improve helm install yaml overrides text box
- onEnterOverrides unglyness needed due to..
  - size of textbox not autosizing properly
  - visuals of textbox without focus not so nice

Commit:
Unit test fix

richard-cox committed:
fetchHelmReleaseFromKubernetes --> fetchHelmReleaseServiceFromKubernetes

richard-cox committed:
Update kube service info when refreshing helm release service list

richard-cox committed:
Add service link in helm release services list
- will not show as link in other services lists

Commit:
Fix sub header title with url param

Commit:
Merge branch 'susecon-tidy-up' into susecon

* susecon-tidy-up:
  Fix sub header title with url param
  Ensure all padding is consistent with dashboard component
  Tidy up kube summary page

Commit:
Change cf favorite deploy menu item to create

richard-cox committed:
Fix no-margin style name change

Commit:
Fix home page margins

richard-cox committed:
Fix kubeService$ error on start up
- works fine, just the compiler being strict

richard-cox committed:
Add wait for release status to release summary loading indicator
- also unsub from pod data chain

richard-cox committed:
Fix deleting indicator on helm release page

richard-cox committed:
Removed debug console.log

Commit:
Tidy up helm release summary tab

Commit:
Merge branch 'create-app-base-step' of https://github.com/cloudfoundry-incubator/stratos into susecon

* 'create-app-base-step' of https://github.com/cloudfoundry-incubator/stratos:
  Removed fdescribe
  Fix e2e tests
  Tidy up types file
  Add specific help text to deploy step
  Add specific deploy types in base tile selector
  Ensure you can't press cancel on endpoint connect dialog while it's busy
  Change cf favorite deploy menu item to create
  Unit test fix
  Unit test fix
  Unit test fixes
  E2e fixes
  CC Fixes
  Add base step to application creation.
  Status updates

Commit:
Make the kube summary layout reusable and apply to helm release summary

Commit:
Improve helm views incluing release and chart summary

Commit:
Missed file updates

Commit:
Added theme for entity title summary component

Commit:
Tighten up the side nav and revert hover change

Commit:
Added icon mode to new nav

Commit:
Fix mobile nav closing

Commit:
Impove sidenav auto hiding

Commit:
Fix side nav on firefox

Commit:
Merge branch 'v2-master' into susecon-tidyup-2

* v2-master:
  Add smoother tile layout breakpoints (once nav side bar changes come in there should be no cut off tile)
  Update GKE given subtype changes
  Fix merge issues
  Fix unit tests
  Fix after merge

Commit:
Improve sidenav styling

Commit:
Overflow fix for nav top section

richard-cox committed:
Small tidies/fixes

richard-cox committed:
helm install reduxing

richard-cox committed:
Remove HelmReleaseLinkComponent and use cellDefinition link config

richard-cox committed:
Remove unused HelmRepositoryCountComponent

richard-cox committed:
Improve typing

richard-cox committed:
fix imports

richard-cox committed:
Use common page size with defaultHelmKubeListPageSize

richard-cox committed:
small tidy ups

richard-cox committed:
Small fixes, cater for endpoints that don't require connecting when showing 'no connected' snackbard

richard-cox committed:
Ensure health check respects doesNotSupportConnect

richard-cox committed:
more minor tidy ups

richard-cox committed:
Use app-no-content-message in release notes & values pages

richard-cox committed:
Fix and use common `unique` function for helm entities & data sources

richard-cox committed:
minor tidyups

richard-cox committed:
More fixes
- ensure all helm & kube entity & row unique functions are correct
- update some lists to use valuePath instead of getValue
- typo fixes
- note on dashboard & margin bug
- comment out some todos

richard-cox committed:
Improve kube typing and use common getKubeAPIResourceGuid

richard-cox committed:
Re-add shared connection icon to endpoint name
- TODO: add to endpoint card

richard-cox committed:
Show shared connection notification in endpoint card

richard-cox committed:
Fix endpoint checking label

richard-cox committed:
Show cancel/close button on connect endpoint dialog
- if the cf is dodgy and not responding the user should be able to exit the modal

richard-cox committed:
Fix lint errors

richard-cox committed:
Fix test compile

richard-cox committed:
Fix unit tests - Round 1

richard-cox committed:
Avoid close-->cancel blip on successful connection

richard-cox committed:
Fix unit tests - Round 2

richard-cox committed:
Fix unit tests - Round 3

richard-cox committed:
Fix unit tests - Round 4

richard-cox commented on src/jetstream/main.go:
```diff
@@ -643,7 +649,7 @@ func start(config interfaces.PortalConfig, p *portalProxy, addSetupMiddleware *s
 	}))
 	e.Use(middleware.SecureWithConfig(middleware.SecureConfig{
-		XFrameOptions: "DENY",
+		XFrameOptions: "sameorigin",
```
Do we also need to make the same change in nginx world?
- deploy/stratos-ui-release/jobs/frontend/templates/nginx.conf.erb
- deploy/containers/nginx/conf/nginx.k8s.conf
- deploy/containers/nginx/conf/nginx.dev.conf

richard-cox commented on src/jetstream/load_plugins.go:
```diff
@@ -1,13 +1,15 @@
 
 import (
-	"github.com/cloudfoundry-incubator/stratos/src/jetstream/plugins/caasp"
+	//	"github.com/cloudfoundry-incubator/stratos/src/jetstream/plugins/caasp"
```
Is the whole caasp plugin not needed anymore?

richard-cox commented on src/jetstream/load_plugins.go:
```diff
@@ -34,8 +36,10 @@ func (pp *portalProxy) loadPlugins() {
-		{"caasp", caasp.Init},
+		{"monocular", monocular.Init},
 		{"userfavorites", userfavorites.Init},
+		{"demo", demo.Init},
```
How is this disabled, and should it be off by default?

richard-cox commented on src/jetstream/plugins/kubernetes/kube_dashboard.go:
```diff
@@ -0,0 +1,365 @@
+	k8s_api_v1 "k8s.io/kubernetes/pkg/apis/core/v1"
+)
+
+// GET /api/v1/namespaces/{namespace}/pods/{name}/proxy
```
Quite a few comments in this file, probably not needed now?

richard-cox committed:
Removed sanitise todo, which happens in [innerHTML]

richard-cox committed:
Bump go version to 1.12 in travis tests

richard-cox commented on src/jetstream/plugins/monocular/main.go:
```diff
@@ -0,0 +1,231 @@
+
+	m.InitSync()
+
+	// u1, _ := url.Parse("https://cloudfoundry-incubator.github.io/stratos")
```
Should this all be nuked?

richard-cox commented on src/jetstream/plugins/monocular/main.go:
```diff
@@ -0,0 +1,231 @@
+	return false
+}
+
+func (m *Monocular) ConfigureMonocular() error {
```
Should now be nuked

richard-cox commented on src/jetstream/repository/interfaces/plugin.go:
```diff
@@ -22,3 +22,11 @@ var JetstreamConfigPlugins []JetstreamConfigInit
 	JetstreamConfigPlugins = append(JetstreamConfigPlugins, plugin)
 }
+
+// JetstreamEndpointNotification is the function signature for a function to call when an endpoint notifcatikon occurs
```
Typo

richard-cox commented on src/jetstream/repository/interfaces/plugin.go:
```diff
@@ -22,3 +22,11 @@ var JetstreamConfigPlugins []JetstreamConfigInit
 }
+
+// JetstreamEndpointNotification is the function signature for a function to call when an endpoint notifcatikon occurs
+type JetstreamEndpointNotification func(EndpointAction, *CNSIRecord)
```
couldn't see where this was used, could be nuked

richard-cox reviewed (commented):
Still todo
- TODO's
- backend unit tests (see https://travis-ci.com/SUSE/stratos/jobs/193382598)
- e2e tests
- themeing
- remove/hide demo button
- disable helm (by not providing helm endpoint config)

nwmac committed:
Add plugin info to the info call

richard-cox committed:
Address/update todos

richard-cox committed:
Address review comments

Commit:
Remove reuse component override on monocular page

Commit:
Fix no margin class

Commit:
Ensure we don't try to fetch sub nav items too many times

Commit:
Remove console log

richard-cox committed:
Disable demo menu action if plugin is disabled

richard-cox committed:
Disable helm (see #150)

richard-cox committed:
Remove unused uiFullView

Commit:
Fix side nav help

Commit:
Fix sidenav help overflow

richard-cox committed:
Disable kube dashboard functionality via plugin config setting
- plugin config - `kubeDashboardEnabled`
- ensure we don't check to see if dashboard installed in kube if dashboard functionality is disabled
- fix kube service ports (can be used in kube namespace servies list even without helm module)

richard-cox committed:
Fix lint issues

richard-cox committed:
e2e fixes - Changes to reflect tabs to side and new sub header with action items

richard-cox committed:
Fix locator

richard-cox committed:
More specific selector

richard-cox committed:
Merge branch 'merge-upstream' into susecon-tidyup-2

richard-cox committed:
Merge fixes - NOTE - help slide out does not block content underneath

richard-cox committed:
Merge branch 'merge-upstream' into susecon-tidyup-2

richard-cox committed:
Fix e2e tests

richard-cox committed:
Fix kube summary icon bleed & unit tests

richard-cox committed:
Merge branch 'merge-upstream' into susecon-tidyup-2

richard-cox committed:
Fixes after merge

richard-cox committed:
Fix new unit test failure #1

richard-cox committed:
Fix new unit test failure #2

richard-cox committed:
Fix new unit test failure #3

richard-cox committed:
Fix new unit test failure #4

richard-cox committed:
Fix failing backend tests

richard-cox committed:
Merge remote-tracking branch 'upstream/v2-master' into susecon-tidyup-2

richard-cox committed:
Changes after merge

richard-cox committed:
Merge remote-tracking branch 'origin/v2-master' into susecon-tidyup-2

richard-cox committed:
Changes after merge
